src/pages/Register/Register.test.js

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import * as rtl from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import {RegisterForm} from "./Register";
+
+const mockRegister = jest.fn().mockResolvedValue(10);
+let wrapper;
+beforeEach(() => {
+  wrapper = rtl.render(<RegisterForm registerNewUser={mockRegister} />);
+});
+
+const FormContainer = () => {
+  return wrapper.queryByTestId("test_register_container");
+};
+
+const Form = () => {
+  return wrapper.queryByTestId("test_register_form");
+};
+
+const EmailInput = () => {
+  return wrapper.queryByTestId("test_email_input");
+};
+
+const UsernameInput = () => {
+  return wrapper.queryByTestId("test_username_input");
+};
+
+const PasswordInput = () => {
+  return wrapper.queryByTestId("test_password_input");
+};
+
+const Button = () => {
+  return wrapper.queryByTestId("test_submit_button");
+};
+
+const EmailWarning = () => {
+  return wrapper.queryByText("Not a valid email");
+};
+
+const UsernameWarning = () => {
+  return wrapper.queryByText("Username must be at least 5 characters");
+};
+
+const PasswordWarning = () => {
+  return wrapper.queryByText("Password must be at least 5 characters");
+};
+
+it("renders without crashing", () => {
+  expect(wrapper.container).toMatchSnapshot();
+});
+
+it("renders all expected elements", () => {
+  expect(FormContainer()).toBeInTheDocument();
+  expect(Form()).toBeInTheDocument();
+  expect(EmailInput()).toBeInTheDocument();
+  expect(UsernameInput()).toBeInTheDocument();
+  expect(PasswordInput()).toBeInTheDocument();
+  expect(Button()).toBeInTheDocument();
+});
+
+it("does not submit due to submit button being disabled", () => {
+  rtl.fireEvent.click(Button());
+  expect(mockRegister).toHaveBeenCalledTimes(0);
+});
+
+it("does not make an axios call if some input fields are left blank", () => {
+  rtl.fireEvent.submit(Form());
+  expect(mockRegister).toHaveBeenCalledTimes(0);
+});
```

Comment:
Your button disable is commented out so those 2 tests are probably not doing what is expected